wgsl/index.bs

```diff
@@ -3598,6 +3583,61 @@ Variables with non-overlapping lifetimes may reuse the memory of previous
 variables; however, new instances of the same variable are not guaranteed to
 use the same memory.
 
+## Variable and Value Declaration Grammar Summary ## {#var-and-value-decl-grammar}
+
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>variable_statement</dfn> :
+
+    | [=syntax/variable_decl=]
+
+    | [=syntax/variable_decl=] [=syntax/equal=] [=syntax/expression=]
+
+    | [=syntax/let=] ( [=syntax/ident=] | [=syntax/variable_ident_decl=] ) [=syntax/equal=] [=syntax/expression=]
+</div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>variable_decl</dfn> :
+
+    | [=syntax/var=] [=syntax/variable_qualifier=] ? ( [=syntax/ident=] | [=syntax/variable_ident_decl=] )
+</div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>variable_ident_decl</dfn> :
+
+    | [=syntax/ident=] [=syntax/colon=] [=syntax/type_decl=]
+</div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>variable_qualifier</dfn> :
+
+    | [=syntax/less_than=] [=syntax/storage_class=] ( [=syntax/comma=] [=syntax/access_mode=] ) ? [=syntax/greater_than=]
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>global_variable_decl</dfn> :
+
+    | [=syntax/attribute_list=] * [=syntax/variable_decl=] ( [=syntax/equal=] [=syntax/const_expression=] ) ?
+</div>
+
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>global_constant_decl</dfn> :
+
+    | [=syntax/let=] ( [=syntax/ident=] | [=syntax/variable_ident_decl=] ) [=syntax/global_const_initializer=] ?
+
+    | [=syntax/attribute_list=] * [=syntax/override=] [=syntax/ident=] [=syntax/equal=] [=syntax/expression=]
```

Comment:
Please combine this rule and the next so that we can know which rule to follow without too much lookahead.

This may shift responsibility to post-parsing checks, e.g. for checking optionality of the initializer.